src/dune_rules/install_rules.ml

```diff
@@ -136,24 +136,30 @@ end = struct
     let module_files =
       let virtual_library = Library.is_virtual lib in
       List.concat_map installable_modules ~f:(fun m ->
-          let cm_file_unsafe kind =
-            Obj_dir.Module.cm_file_unsafe obj_dir m ~kind
+          let visibility = Module.visibility m in
+          let cmi_file = Obj_dir.Module.cm_file_unsafe obj_dir m ~kind:Cmi in
+          let cm_file kind = Obj_dir.Module.cm_file obj_dir m ~kind in
+          let cmts =
+            List.filter_map Ml_kind.all ~f:(fun ml_kind ->
+                Obj_dir.Module.cmt_file obj_dir m ~ml_kind)
+          in
+          let if_ b x =
```

Comment:
The implementation of `if_` looks backwards, but tests keep failing even if that is fixed.

Reply:
Thanks. Fixed.